src/main/java/graphql/schema/idl/SchemaTypeChecker.java

```diff
@@ -461,6 +461,52 @@ private void checkInterfacesAreImplemented(List<GraphQLError> errors, TypeDefini
         };
     }
 
+    @SuppressWarnings("SimplifiableIfStatement")
+    private boolean isTypeSubTypeOf(TypeDefinitionRegistry registry, Type maybeSubType, Type superType) {
+        TypeInfo maybeSubTypeInfo = TypeInfo.typeInfo(maybeSubType);
+        TypeInfo superTypeInfo = TypeInfo.typeInfo(superType);
+        // Equivalent type is a valid subtype
+        if (maybeSubTypeInfo.equals(superTypeInfo)) {
+            return true;
+        }
+
+
+        // If superType is non-null, maybeSubType must also be non-null.
+        if (superTypeInfo.isNonNull()) {
+            if (maybeSubTypeInfo.isNonNull()) {
+                return isTypeSubTypeOf(registry, maybeSubTypeInfo.unwrapOneType(), superTypeInfo.unwrapOneType());
+            }
+            return false;
+        }
+        if (maybeSubTypeInfo.isNonNull()) {
+            // If superType is nullable, maybeSubType may be non-null or nullable.
+            return isTypeSubTypeOf(registry, maybeSubTypeInfo.unwrapOneType(), superType);
+        }
+
+        // If superType type is a list, maybeSubType type must also be a list.
+        if (superTypeInfo.isList()) {
+            if (maybeSubTypeInfo.isList()) {
+                return isTypeSubTypeOf(registry, maybeSubTypeInfo.unwrapOneType(), superTypeInfo.unwrapOneType());
+            }
+            return false;
+        }
+        if (maybeSubTypeInfo.isList()) {
+            // If superType is not a list, maybeSubType must also be not a list.
+            return false;
+        }
+
+        // If superType type is an abstract type, maybeSubType type may be a currently
+        // possible object type.
```

Comment:
maybe this a spec level problem, but I don't see they point to allow union types here: you can't implement a union type

Reply:
This come almost word for word from graphql-js